app/Http/Controllers/CategoriaController.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Categoria;
+
+class CategoriaController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $categorias =Categoria::all();
+        return $categorias;
+    }
+
+   
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+       $categoria=new Categoria();
+       $categoria->nombre = $request->nombre;
+       $categoria->descripcion = $request->descripcion;
+       $categoria->condicion='1';
+       $categoria->save();
+    }
+
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request)
+    {
+       $categoria=Categoria::findOrFail($request->id);
+       $categoria->nombre = $request->nombre;
+       $categoria->descripcion = $request->descripcion;
+       $categoria->condicion='1';
+       $categoria->save();
+
+    }
+    public function desactivar(Request $request)
```

Comment:
Use only one function to activate/deactivate example
```suggestion
   public function cambiarCondicion(Request $request)
    {
        $categoria=Categoria::findOrFail($request->id);
        $categoria->condicion = !$categoria->condicion;
        $categoria->save();
    }
```